File: my-submissions/1910-check-if-binary-string-has-at-most-one-segment-of-ones/solution.java
class Solution {
    public boolean checkOnesSegment(String s) {
        boolean seenOne = false;
        boolean zeroAfterOne = false;

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (ch == '1') {
                if (zeroAfterOne) {
                    return false;
                }
                seenOne = true;
            } else {
                if (seenOne) {
                    zeroAfterOne = true;
                }
            }
        }
        return true;
    }
}

File: my-submissions/1910-check-if-binary-string-has-at-most-one-segment-of-ones/solution.py
class Solution(object):
    def checkOnesSegment(self, s):
        """
        :type s: str
        :rtype: bool
        """
        seen_one = False
        zero_after_one = False

        for ch in s:
            if ch == '1':
                if zero_after_one:
                    return False
                seen_one = True
            else:
                if seen_one:
                    zero_after_one = True

        return True
